server/src/main/java/io/crate/execution/engine/sort/Sort.java

```diff
@@ -44,26 +52,42 @@ public static <T> CompletableFuture<List<T>> parallelSort(List<T> list,
         int itemsPerThread = list.size() / Math.max(1, numAvailableThreads);
         if (numAvailableThreads <= 1 || itemsPerThread < minItemsPerThread) {
             list.sort(comparator);
+            allocateBytes.accept(sortOverheadBytes(list.size()));
             return completedFuture(list);
         }
         List<List<T>> partitions = Lists.partition(list, itemsPerThread);
-        ArrayList<CompletableFuture<List<T>>> futures = new ArrayList<>(partitions.size());
+        allocateBytes.accept(RamUsageEstimator.sizeOfObject(partitions)); // Accounts for nested list
```

Comment:
Afaik the `partitions` are implemented as a view to the underlaying list, so only the additional `List` instances should be accounted for (shallow size).